lib/OpModel/TTNN/TTNNOpModelLib.cpp

```diff
@@ -0,0 +1,172 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstddef>
+#include <stdexcept>
+#include <tuple>
+
+#include <llvm/Support/Casting.h>
+#include <mlir/IR/AttrTypeSubElements.h>
+
+#include "TTNNOpModel.hpp"
+#include "TTNNOpModelLib_Impl.hpp"
+#include "impl/buffers/buffer.hpp"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttnn/tensor/types.hpp"
```

Comment:
Copy pasting #include standard:

Include style is defined by LLVM coding standards: https://llvm.org/docs/CodingStandards.html#include-style

We prefer these #includes to be listed in this order:

	1. Main Module Header
	2. Local/Private Headers
	3. LLVM project/subproject headers (clang/..., lldb/..., llvm/..., etc)
	4. System #includes

Using `TTIRToTTNN.cpp` as an example, this is what includes would look like for us:

```
#include "ttmlir/Conversion/TTIRToTTNN/TTIRToTTNN.h"  # main header

#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"  # these are local/private headers
#include "ttmlir/Dialect/TTNN/Utils/Utils.h"

#include "mlir/Dialect/MemRef/IR/MemRef.h"  # llvm project/subproj headers
#include "llvm/Support/LogicalResult.h"

#include <cstdio>  # system includes
#include <algorithm>
```

Reply:
thanks for calling this out